Multisig/Logic/BluetoothController.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  BluetoothController.swift
+//  Multisig
+//
+//  Created by Moaaz on 7/26/21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import CoreBluetooth
+
+protocol LedgerNanoXControllerDelegate {
+    func bluetoothControllerDidReceive(response: String)
+    func bluetoothControllerDidFailToConnectBluetooth()
+    func bluetoothControllerDidDiscoverDevice()
+    func bluetoothControllerDidDisconnectDevice()
+}
+
+class BluetoothController: NSObject {
+    static let shared = BluetoothController()
+    private var centralManager: CBCentralManager!
+    var delegate: LedgerNanoXControllerDelegate?
+
+    private var devices: [BluetoothDevice] = []
+    private var supportedDevices: [DeviceConstant] = []
+
+    private var supportedDeviceUuids: [CBUUID] { supportedDevices.compactMap { $0.uuid } }
+    private var supportedDeviceNotifyUuids: [CBUUID] { supportedDevices.compactMap { $0.notifyUuid } }
+
+    override init() {
+        super.init()
+    }
+
+    func scan(devices: [DeviceConstant]) {
+        supportedDevices = devices
+        centralManager = CBCentralManager(delegate: self, queue: nil)
+    }
+
+    func stopScan() {
+        centralManager.stopScan()
+    }
+}
+
+extension BluetoothController: CBCentralManagerDelegate {
+    func centralManagerDidUpdateState(_ central: CBCentralManager) {
+        switch central.state {
+        case .poweredOn:
+            centralManager.scanForPeripherals(withServices: supportedDeviceUuids)
+        default:
+            // Bluetooth might be off or not permitted to use
+            delegate?.bluetoothControllerDidFailToConnectBluetooth()
```

Comment:
should we have an explicit method for a `not permitted` state? I think we should offer a user to enable Bluetooth in this case.

Reply:
we need to pass error message so that user knows how to fix the problem